packages/cardano-services/test/TxWorker/utils.test.ts

```diff
@@ -0,0 +1,131 @@
+import { createDnsResolver, getOgmiosTxSubmitProvider } from '../../src';
+import { createLogger } from 'bunyan';
+import { createMockOgmiosServer } from '../../../ogmios/test/mocks/mockOgmiosServer';
+import { listenPromise, serverClosePromise } from '../../src/util';
+import { ogmiosServerReady } from '../util';
+import http from 'http';
+
+const flushPromises = () => new Promise((resolve) => setImmediate(resolve));
+
+jest.mock('dns', () => ({
+  promises: {
+    resolveSrv: async (serviceName: string): Promise<SrvRecord[]> => {
+      if (serviceName === process.env.OGMIOS_SRV_SERVICE_NAME)
+        return [{ name: 'localhost', port: 1337, priority: 6, weight: 5 }];
+      if (serviceName === process.env.RABBITMQ_SRV_SERVICE_NAME)
+        return [{ name: 'localhost', port: 5672, priority: 6, weight: 5 }];
+      return [];
+    }
+  }
+}));
+
+describe('TxSubmitWorker abstraction', () => {
+  describe('getRunningTxSubmitWorker', () => {
+    let mockServer: http.Server;
+    let connection: Connection;
+    let txSubmitWorker: RunningTxSubmitWorker;
+    let txSubmitProvider: TxSubmitProvider;
+    const ogmiosPortDefault = 1337;
+    const rabbitmqPortDefault = 5672;
+    const logger = createLogger({ level: 'error', name: 'test' });
+    const dnsResolver = createDnsResolver({ factor: 1.1, maxRetryTime: 1000 }, logger);
+    const srvRecord = { name: 'localhost', port: rabbitmqPortDefault, priority: 1, weight: 1 };
+
+    beforeEach(async () => {
+      connection = Ogmios.createConnectionObject({ port: ogmiosPortDefault });
+      mockServer = createMockOgmiosServer({
+        healthCheck: { response: { networkSynchronization: 0.999, success: true } }
+      });
+      await listenPromise(mockServer, connection);
+      await ogmiosServerReady(connection);
+    });
+
+    afterEach(async () => {
+      if (mockServer !== undefined) {
+        await serverClosePromise(mockServer);
+      }
+
+      await txSubmitWorker.stop();
+    });
+
+    it('should instantiate a running worker with service discovery', async () => {
+      const dnsResolverMock = jest.fn().mockResolvedValueOnce(srvRecord);
+      txSubmitProvider = await getOgmiosTxSubmitProvider(dnsResolver, logger, {
+        cacheTtl: 10_000,
+        ogmiosSrvServiceName: process.env.OGMIOS_SRV_SERVICE_NAME,
+        serviceDiscoveryBackoffFactor: 1.1,
+        serviceDiscoveryTimeout: 1000
+      });
+
+      txSubmitWorker = await getRunningTxSubmitWorker(dnsResolverMock, txSubmitProvider, logger, {
+        cacheTtl: 10_000,
+        rabbitmqSrvServiceName: process.env.RABBITMQ_SRV_SERVICE_NAME,
+        serviceDiscoveryBackoffFactor: 1.1,
+        serviceDiscoveryTimeout: 1000
+      });
+
+      await expect(dnsResolverMock).toBeCalledTimes(1);
+      expect(txSubmitWorker.getStatus()).toEqual('connected');
+    });
+  });
+
+  describe('startTxSubmitWorkerWithDiscovery', () => {
+    it('starts and stops the worker', async () => {
```

Comment:
This is a bit misleading, as the test reads "startTxSubmitWorkerWithDiscovery starts and stops the worker". 
```suggestion
    it('returns a started worker, which can then be stopped', async () => {
```

Reply:
https://github.com/input-output-hk/cardano-js-sdk/pull/344/commits/89a1a0d3cf617df8746f6493e70edfa219226835